ext/magic/mkmf-upstream.rb

```diff
@@ -0,0 +1,2881 @@
+          refs.map {|n|"    printf(\"%d\", (*#{n}p)());\n"}.join("") +
+          "  }\n"
+      end
+    end
+    src
+  end
+
+  extend self
+  init_mkmf
+
+  $make = with_config("make-prog", ENV["MAKE"] || "make")
+  make, = Shellwords.shellwords($make)
+  $nmake = nil
+  case
+  when $mswin
+    $nmake = ?m if /nmake/i =~ make
+  end
+  $ignore_error = $nmake ? '' : ' 2> /dev/null || true'
+
+  RbConfig::CONFIG["srcdir"] = CONFIG["srcdir"] =
+    $srcdir = arg_config("--srcdir", File.dirname($0))
+  $configure_args["--topsrcdir"] ||= $srcdir
+  if $curdir = arg_config("--curdir")
+    RbConfig.expand(curdir = $curdir.dup)
+  else
+    curdir = $curdir = "."
+  end
+  unless File.expand_path(RbConfig::CONFIG["topdir"]) == File.expand_path(curdir)
+    CONFIG["topdir"] = $curdir
+    RbConfig::CONFIG["topdir"] = curdir
+  end
+  $configure_args["--topdir"] ||= $curdir
+  $ruby = arg_config("--ruby", File.join(RbConfig::CONFIG["bindir"], CONFIG["ruby_install_name"]))
+
+  RbConfig.expand(CONFIG["RUBY_SO_NAME"])
+
+  # :startdoc:
+
+  split = Shellwords.method(:shellwords).to_proc
+
+  EXPORT_PREFIX = config_string('EXPORT_PREFIX') {|s| s.strip}
+
+  hdr = ['#include "ruby.h"' "\n"]
+  config_string('COMMON_MACROS') do |s|
+    Shellwords.shellwords(s).each do |w|
+      w, v = w.split(/=/, 2)
+      hdr << "#ifndef #{w}"
+      hdr << "#define #{[w, v].compact.join(" ")}"
+      hdr << "#endif /* #{w} */"
+    end
+  end
+  config_string('COMMON_HEADERS') do |s|
+    Shellwords.shellwords(s).each {|w| hdr << "#include <#{w}>"}
+  end
+
+  ##
+  # Common headers for Ruby C extensions
+
+  COMMON_HEADERS = hdr.join("\n")
+
+  ##
+  # Common libraries for Ruby C extensions
+
+  COMMON_LIBS = config_string('COMMON_LIBS', &split) || []
+
+  ##
+  # make compile rules
+
+  COMPILE_RULES = config_string('COMPILE_RULES', &split) || %w[.%s.%s:]
+  RULE_SUBST = config_string('RULE_SUBST')
+
+  ##
+  # Command which will compile C files in the generated Makefile
+
+  COMPILE_C = config_string('COMPILE_C') || '$(CC) $(INCFLAGS) $(CPPFLAGS) $(CFLAGS) $(COUTFLAG)$@ -c $(CSRCFLAG)$<'
+
+  ##
+  # Command which will compile C++ files in the generated Makefile
+
+  COMPILE_CXX = config_string('COMPILE_CXX') || '$(CXX) $(INCFLAGS) $(CPPFLAGS) $(CXXFLAGS) $(COUTFLAG)$@ -c $(CSRCFLAG)$<'
+
+  ##
+  # Command which will translate C files to assembler sources in the generated Makefile
+
+  ASSEMBLE_C = config_string('ASSEMBLE_C') || COMPILE_C.sub(/(?<=\s)-c(?=\s)/, '-S')
+
+  ##
+  # Command which will translate C++ files to assembler sources in the generated Makefile
+
+  ASSEMBLE_CXX = config_string('ASSEMBLE_CXX') || COMPILE_CXX.sub(/(?<=\s)-c(?=\s)/, '-S')
+
+  ##
+  # Command which will compile a program in order to test linking a library
+
+  TRY_LINK = config_string('TRY_LINK') ||
+    "$(CC) #{OUTFLAG}#{CONFTEST}#{$EXEEXT} $(INCFLAGS) $(CPPFLAGS) " \
+    "$(CFLAGS) $(src) $(LIBPATH) $(LDFLAGS) $(ARCH_FLAG) $(LOCAL_LIBS) $(LIBS)"
```

Comment:
This line is patched in rake-compiler-dock [here](https://github.com/rake-compiler/rake-compiler-dock/blob/3ab8a950920f19b75479c195a77291e9ada7fa8b/Dockerfile.mri.erb#L210) to include `-lruby-static`. So the build failure is due to using this custom `mkmf.rb`. Can the line changed here as well? Then the workaround of `have_library('ruby-static')` is no longer necessary.